lib/upsert/merge_function/postgresql.rb

```diff
@@ -73,16 +75,20 @@ def pg_function(row)
               first_try = false
               create!
               retry
-            else
-              Upsert.logger.info %{[upsert] Failed to create function #{name.inspect} for some reason}
-              raise pg_error
             end
+            Upsert.logger.info %{[upsert] Failed to create function #{name.inspect} for some reason}
+            raise pg_error
           else
             raise pg_error
           end
         end
       end
 
+      def function_exists?
+        # The ::int is a hack until jruby+jdbc is happy with bigints being returned
+        @function_exists ||= controller.connection.execute("SELECT count(*)::int AS cnt FROM pg_proc WHERE lower(proname) = lower('#{name}')").first["cnt"].to_i > 0
```

Comment:
Right :)

That being said, do we want to memoize the false value?  It may be beneficial to re-check whether or not the function exists so we don't attempt to re-create it (I'm thinking of possible race conditions here).